src/transports/smart_pkt.c

```diff
@@ -309,8 +311,8 @@ static int ng_pkt(git_pkt **out, const char *line, size_t len)
 	if (len < 1)
 		goto out_err;
 	line = ptr + 1;
-	len -= 1;
-	if (!(ptr = memchr(line, '\n', len)))
+
+	if (!(ptr = memchr(line, '\n', eol - line)))
```

Comment:
I think I prefer your `eol`-pointer thing. It reliefs us from always updating the `len` field, and thus is less error-prone.